### DISCUSSION TITLE
Question about using i18n

### DISCUSSION THREAD

**Discussion by CleitonDeLima** [codingjoe/django-select2#51] [Q&A] [answered]
I have been using this library for a long time, it is the best option for autocomplete fields in django, thank you @codingjoe for keeping this project alive.

In my project I had a problem using the `django-select2` with i18n. In time I will report my problem here, I believe it is a bug.
I didn't create an issue because  `django.contrib.admin` without lib `django-select2` has the same behavior.

The language I use in my project is Brazilian (`pt-BR`)
I believe that in django when choosing `LANGUAGE_CODE` as `pt-br` or `pt-BR` there is no difference in translation.

Use cases:
[Code location](https://github.com/codingjoe/django-select2/blob/master/django_select2/forms.py#L114-L120)

```python
i18n_name = SELECT2_TRANSLATIONS.get(lang)
if i18n_name not in settings.SELECT2_I18N_AVAILABLE_LANGUAGES:
    i18n_name = None

i18n_file = (
    ("%s/%s.js" % (settings.SELECT2_I18N_PATH, i18n_name),) if i18n_name else ()
)
```

- Using `pt-br`
```python 
# settings.py
LANGUAGE_CODE="pt-br"
```

> `i18n_name` is found 
> `SELECT2_TRANSLATIONS` are all lowercase, configuration coming from `django.contrib.admin`.
> `i18n_file` is not found, because `SELECT2_I18N_AVAILABLE_LANGUAGES` contains `pt-BR`. [here](https://github.com/codingjoe/django-select2/blob/master/django_select2/conf.py#L150)

- Using `pt-BR`
```python 
# settings.py
LANGUAGE_CODE="pt-BR"
```
> `i18n_name` and  `i18n_file` not found.

My solution I have in the project is as follows:
```python
LANGUAGE_CODE="pt-br"
```
and html tag uses fixed configuration for select2 to work and not `{{ LANGUAGE_CODE }}`:

```html
<html lang="pt-BR">
...
</html>
```
Can this be a bug?

**Comment by codingjoe** [comment, marked as answer]
Hi @CleitonDeLima,

Thank you for your kind words. And good point, that could be improved both in this package and in Django itself. If you open a bug with Django, please add me to cc, so I can do triage and provide a patch there too. Thanks!

Best,
Joe

**Reply by CleitonDeLima** [reply]
https://code.djangoproject.com/ticket/33070
